### PR TITLE
Added 1.62 chunk migrations

### PR DESCRIPTION
**Pull request by GSterbrant**
Added chunk migrations for the frontend/backend shader changes coming in 1.62.

image

### CODE REVIEW

File: content/en/user-manual/graphics/shader-chunk-migrations.md [hunk 1]
```diff
@@ -29,6 +29,93 @@ By doing this you will no longer see warning messages in the console.
 
 The following tables break down the chunk changes by Engine release.
 
+#### *Engine v1.62*
+
+In 1.62, almost all backend chunks have been changed to accomodate for the more explicit frontend-backend split of the shaders. This means that any custom backend shader chunks must move away from using globals to using the arguments passed to them by the shader backend.
```
Comment (yaustar) [inline content/en/user-manual/graphics/shader-chunk-migrations.md:34]: We may have to explain what backend/frontend chunks are. We've not shared what this means

File: content/en/user-manual/graphics/shader-chunk-migrations.md [hunk 2]
```diff
@@ -41,7 +128,7 @@ The following tables break down the chunk changes by Engine release.
 
 ---
 
-#### Engine v1.57
+#### *Engine v1.57*
```
Comment (yaustar) [inline content/en/user-manual/graphics/shader-chunk-migrations.md:131]: Do we want to bold all titles in this page 😅 ?

File: content/en/user-manual/graphics/shader-chunk-migrations.md (hunk 1 above)
Comment (GSterbrant) [reply content/en/user-manual/graphics/shader-chunk-migrations.md]: Good point, done.

File: content/en/user-manual/graphics/shader-chunk-migrations.md [hunk 3]
```diff
@@ -29,6 +29,95 @@ By doing this you will no longer see warning messages in the console.
 
 The following tables break down the chunk changes by Engine release.
 
+#### *Engine v1.62*
+
+In PlayCanvas, we have two sets of shader chunks, one set we refer to as the shader frontend, which provide values for the arguments passed to our lighting algorithm, also called the shader backend. With 1.62, we are creating a clearer distinction between these two, such that the values passed to the backend are well defined and known in advance, not automatically generated. This allows for writing a fully custom shader that can interface with our lighting code just like how our native materials do.
```
Comment (yaustar) [inline content/en/user-manual/graphics/shader-chunk-migrations.md:34]:
```suggestion
In PlayCanvas, we have two sets of shader chunks, one set we refer to as the shader frontend, which provide values for the arguments passed to our lighting algorithm, also called the shader backend. 

With 1.62, we are creating a clearer distinction between these two, such that the values passed to the backend are well defined and known in advance, not automatically generated. This allows for writing a fully custom shader that can interface with our lighting code just like how our native materials do. 
```

File: content/en/user-manual/graphics/shader-chunk-migrations.md (hunk 2 above)
Comment (GSterbrant) [reply content/en/user-manual/graphics/shader-chunk-migrations.md]: I removed the italics, and increased the Engine headers as they were almost undetectable.

File: content/en/user-manual/graphics/shader-chunk-migrations.md [hunk 4]
```diff
@@ -29,9 +29,101 @@ By doing this you will no longer see warning messages in the console.
...
+| `metalnessModulate` | <ul><li>Accepts a `LitShaderArguments` struct which is updated by the chunk. Removes the reliance on `dSpecularity`, `dMetalness` and `dAlbedo`</li></ul>
+| `output(Alpha/AlphaPremul)` | <ul><li>Uses `litShaderArgs.opacity` instead of `dAlpha`</li></ul>
+| `reflDir(Aniso)` | <ul><li>Accepts a vec3 world normal and a float value for gloss, instead of using `dGlossiness` and `dNormalW`</li></ul>
+| `reflection(CC/Cube/Env/EnvHQ/Sphere/SphereLow)` | <ul><li>Accepts a vec3 reflection direction and a float gloss value instead of using `dReflDirW`/`ccReflDirW` and `dGlossiness`</li></ul>
+| `reflectionSheen` | <ul><li>Accepts a vec3 world normal and a float gloss value instead of using `dNormalW` and `sGlossiness`</li></ul>
+| `refraction(Cube/Dynamic)` | <ul><li>Accepts a vec3 world normal, float thickness, float gloss, vec3 specularity, vec3 albedo, float transmission and `IridescenceArgs` instead of using `dNormalW`, `dAlbedo`, `dTransmission`, `dThickness`, `dGlossiness`, `dSpecularity` and passes the iridescence arguments to the fresnel function</li></ul>
```
Comment (willeastcott) [inline content/en/user-manual/graphics/shader-chunk-migrations.md:121]: Looks like each table row is missing the trailing `|`

Comment (mvaligursky) [reply content/en/user-manual/graphics/shader-chunk-migrations.md:121]: Could you please build the manual locally, and attach the screenshot of the new section in the PR description.

Comment (willeastcott) [reply content/en/user-manual/graphics/shader-chunk-migrations.md]: Looks like the trailing `|` is still missing.

**Pull request merged**